src/Generator/TypeHintedMethodGenerator.php

```diff
@@ -45,14 +45,7 @@ public function __construct(ConsoleIO $io, TemplateRenderer $templates, Filesyst
         $this->filesystem = $filesystem ?: new Filesystem;
     }
 
-    /**
-     * @param Resource $resource
-     * @param string   $generation
-     * @param array    $data
-     *
-     * @return bool
-     */
-    public function supports(Resource $resource, $generation, array $data)
+    public function supports(Resource $resource, string $generation, array $data) : bool
```

Comment:
because you've added these hints, the php requirement in `composer.json` needs to be increased